### ISSUE TITLE
Should 'start' CPO require that argument is a non-const lvalue?

### ISSUE THREAD

**Issue by lewissbaker** [cplusplus/sender-receiver#273]
Extracting the remaining issue from #100.

The `start()` CPO specification (as of P2300R10) says that it is ill-formed to pass an rvalue and otherwise `start(op)` is equivalent to `MANDATE-NOTHROW(op.start())`.

However, this permits `op` to be a const-lvalue if the operation-state object has a const-qualified `.start()` member-function.

Do we want to allow this?
Or do we want to instead specify that passing anything other than an unqualified lvalue to `start()` is ill-formed?

**Comment by ericniebler** [comment]
what would be the reason to require (or mandate) that `.start()` cannot be called on a `const` operation state?

**Comment by lewissbaker** [comment]
The only reason we can see for restricting calling `start()` on a const-lvalue would be to reserve the right to assign different semantics for this in future by preventing existing code from using this pattern.

However, there is no clear use-case for supporting const-qualified operation-states that I can imagine. And even if we did come up with some use-case/new-semantic we want to describe here, we can always just add a new method-name instead as a more refined concept.

Closing as "not worth spending more time on at this stage".
